src/public/js/util.js

```diff
@@ -112,39 +124,101 @@ var UTIL = (function() {
      0x08, 0x4e, 0xd7, 0xe3, 0x5d, 0x50, 0x1e, 0xb3,
      0x5b, 0x23, 0x38, 0x34, 0x68, 0x46, 0x03, 0x8c,
      0xdd, 0x9c, 0x7d, 0xa0, 0xcd, 0x1a, 0x41, 0x1c]);
+  multInv.get = function(x) {
+    if (x < 1 || x >= multInv.length) {
+      throw new Error('Invalid lookup into multInv LUT.');
+    }
+    return multInv[x];
+  };
+
+  // Returns product of two numbers in GF(256).
+  // mult1 and mult2 should both be integers between 0 and 255 inclusive.
+  function fieldMult(mult1, mult2) {
+    if (mult1 == 0 || mult2 == 0) {
+      return 0;
+    }
+    var t = log3.get(mult1) + log3.get(mult2);
+    if (t > 255) {
+      t = t - 255;
+    }
+    return exp3.get(t);
+  }
 
-  function fieldDivide(dividend, divisor) {
+  // Returns quotient of two numbers in GF(256).
+  // dividend and divisor should both be integers between 0 and 255 inclusive.
+  function fieldDiv(dividend, divisor) {
     if (divisor == 0) {
-      throw new Error('Can\'t divide by 0');
+      throw new Error('Can\'t divide by 0.');
     } else if (dividend == 0) {
       return 0;
     }
-    var t = log3[dividend] - log3[divisor];
+    var t = log3.get(dividend) - log3.get(divisor);
     if (t < 0) {
       t = t + 255;
     }
-    return exp3[t];
+    return exp3.get(t);
   }
 
-  function fieldMultiply(mult1, mult2) {
-    if (mult1 == 0 || mult2 == 0) {
-      return 0;
-    }
-    var t = log3[mult1] + log3[mult2];
-    if (t > 255) {
-      t = t - 255;
+  // Returns sum of two polynomials in GF(256).
+  // Both addend1 and addend2 are big-endian arrays of integers which
+  // represent polynomials (so, [1,2,3] is x^2 + 2*x + 3)
+  function polynomialAdd(addend1, addend2) {
+    var resultLength = Math.max(addend1.length, addend2.length);
+    var result = new Uint8Array(resultLength);
+    var digit;
+    for (var i = 0; i < resultLength; i++) {
+      digit = 0;
+      if (i < addend1.length) {
+        digit ^= addend1[addend1.length - 1 - i];
+      }
+      if (i < addend2.length) {
+        digit ^= addend2[addend2.length - 1 - i];
+      }
+      result[resultLength - 1 - i] = digit;
     }
-    return exp3[t];
+    return result;
   }
 
-  function shift(msg, rot) {
-    return msg.split('').reduce(function(accumulator, letter, index) {
-      var shiftedCharCode = letter.charCodeAt() + rot;
-      var shiftedLetter = String.fromCharCode(shiftedCharCode);
-      return accumulator + shiftedLetter;
-    }, '');
+  // Helper function for polynomial multiplication
+  function isZero(n) {
+    return n == 0;
   }
 
+  // Returns product of two polynomials in GF(256).
+  // Both mult1 and mult2 are big-endian arrays of integers which
+  // represent polynomials (so, [1,2,3] is x^2 + 2*x + 3)
+  function polynomialMult(mult1, mult2) {
+    // Handle the zero case
+    if (mult1.every(isZero) || mult2.every(isZero)) {
```

Comment:
did it a bit differently so that all of the polynomial operations that return polynomials return arrays with no leading zeros -- lmk what you think